src/jdk.incubator.vector/share/classes/jdk/incubator/vector/VectorShuffle.java

```diff
@@ -523,6 +557,24 @@ private static int unzipIndex(int i, int vlen, int part) {
      */
     public abstract void intoArray(int[] a, int offset);
 
+    /**
+     * Stores this shuffle into a {@link MemorySegment} starting at offset.
+     * <p>
+     * For each shuffle lane {@code N}, the lane source index
+     * stored for that lane element is stored into the
+     * {@link MemorySegment} at {@code offset + N * 4}.
```

Comment:
Similar here we need to say a little more e.g.,

> that lane element is written into the {@link MemorySegment) at ... using an {@link ValueLayout#JAVA_INT_UNALIGNED unaligned `int`} layout.

Also the segment might be read only